addons/event/report/event_event_templates.xml

```diff
@@ -50,6 +50,19 @@
                     </div>
                     <div class="o_event_foldable_badge_barcode mt-2">
                         <div class="o_event_foldable_badge_barcode_container">
+                            <div class="col-12 text-center">
+                                <div class="pt-3 mt-2 pb-3">
+                                    <t t-if="env['ir.config_parameter'].sudo().get_param('event.use_event_barcode') == 'True'">
```

Comment:
I would make a use_barcode field on event / registration models, that hides the low-level detail of checking the parameter check. That way you simply do "t-if='attendee.use_barcode' ..." and voila.